app/models/payment.rb

```diff
@@ -7,11 +7,28 @@ class Payment < ApplicationRecord
   validates :payroll_reference, :gross_value, :national_insurance, :employers_national_insurance, :tax, :net_pay, :gross_pay, presence: true, on: :upload
   validates :gross_value, :national_insurance, :employers_national_insurance, :student_loan_repayment, :tax, :net_pay, :gross_pay, numericality: true, allow_nil: true
 
-  delegate :first_name, :middle_name, :surname, :national_insurance_number, :payroll_gender, :date_of_birth, :email_address, :address_line_1, :address_line_2, :address_line_3, :address_line_4, :postcode, :has_student_loan, :student_loan_plan, :banking_name, :bank_sort_code, :bank_account_number, :building_society_roll_number, to: :claim_for_personal_details
+  validate :personal_details_must_be_consistent
+
+  PERSONAL_DETAILS_ATTRIBUTES_PERMITTING_DISCREPANCIES = %i[first_name middle_name surname national_insurance_number payroll_gender email_address address_line_1 address_line_2 address_line_3 address_line_4 postcode has_student_loan banking_name]
+  PERSONAL_DETAILS_ATTRIBUTES_FORBIDDING_DISCREPANCIES = %i[national_insurance_number date_of_birth student_loan_plan bank_sort_code bank_account_number building_society_roll_number]
+
+  delegate(*(PERSONAL_DETAILS_ATTRIBUTES_PERMITTING_DISCREPANCIES + PERSONAL_DETAILS_ATTRIBUTES_FORBIDDING_DISCREPANCIES), to: :claim_for_personal_details)
 
   private
 
+  def personal_details_must_be_consistent
+    mismatching_attributes = PERSONAL_DETAILS_ATTRIBUTES_FORBIDDING_DISCREPANCIES.select { |attribute|
+      claims.map(&attribute).uniq.count > 1
+    }
+
+    if mismatching_attributes.any?
+      claims_sentence = claims.map(&:reference).to_sentence
+      attributes_sentence = mismatching_attributes.map { |attribute| Claim.human_attribute_name(attribute).downcase }.to_sentence
+      errors.add(:claims, "#{claims_sentence} have different values for #{attributes_sentence}")
+    end
+  end
+
   def claim_for_personal_details
-    @claim_for_personal_details ||= claims.first
+    @claim_for_personal_details ||= claims.max_by(&:created_at)
```

Comment:
We can lean on ActiveRecord for this:

```suggestion
    @claim_for_personal_details ||= claims.last
```

Reply:
That was what I originally wrote, but it didn't work when I set up the payment like this in the tests:

```ruby
let(:claims) do
  [
    build(:claim, :approved, first_name: "Margaret", address_line_1: "17 Green Road", payroll_gender: :female, created_at: 5.days.ago),
    build(:claim, :approved, first_name: "John", address_line_1: "64 West Lane", payroll_gender: :male, created_at: 10.days.ago),
  ]
end
```

It returns the claim from `10.days.ago`, which isn't what I want.

Reply:
So the reason using `last` will fail in that spec is because the records aren’t persisted – they’re all new records, so last is falling back to returning the last in the array. Do you think the method will ever be called against unpersisted claims? Also we should probably be returning the most recently submitted claim, i.e. check `submitted_at`.

Reply:
These things about what does and doesn't work for an unsaved relation continue to catch me out! It isn't being used with unpersisted claims at the moment, no. I can't really speculate about how it might be used in the future, though. I'll change it to use `submitted_at` and use persisted records in the specs.

Reply:
I've made that change. Had to change a few other specs to persist their records too, for them to continue working. Does it look alright to you?

Reply:
Yep, working with the persisted records makes sense, new changes look like they cover reality better than before 👍🏽